### PR TITLE
add 7-POST/api/reviews/:review_id/comments

### PR DESCRIPTION
**Pull request by HassanAlhaji**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -116,11 +116,48 @@ describe("GET/api/categories", () => {
...
+      return request(app)
+      .get('/api/reviews/orange/comments')
+      .expect(400)
+      .then((response)=>{
+        const msg = response.body.msg
+        expect(msg).toBe("bad request")
+      })
+ })
+ 
+      test.only('should Responds with the posted comment',()=>{
```
Comment (B7Bprog) [inline __tests__/app.test.js]: Please remove .only to allow all the tests to run and resubmit your pr.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Hassan, please remove the `.only` after one of your tests, run all the tests, make sure they are passing, and then send us the pull request again without any `.only`.

File: __tests__/app.test.js (hunk 1 above)
Comment (HassanAlhaji) [reply __tests__/app.test.js]: sorry i just forgot it now it's done

File: __tests__/app.test.js [hunk 2]
```diff
@@ -116,11 +116,48 @@ describe("GET/api/categories", () => {
   .expect(200)
   .then((response)=>{
     
-    console.log(response.body, 'inside test')
     expect(response.body.comments).toStrictEqual([])
   })
  })
 
+ describe('POST api/reviews/:review_id/comments',()=>{
+    test('400: not found', ()=>{
```
Comment (robert112233) [inline __tests__/app.test.js:124]: 124 needs to be changed to 404, the same way you have it in the expect statement 👍

File: __tests__/app.test.js [hunk 3]
```diff
@@ -116,11 +116,48 @@ describe("GET/api/categories", () => {
...
+    })
+    test('400:invalid review_id',()=>{
+      return request(app)
+      .get('/api/reviews/orange/comments')
+      .expect(400)
+      .then((response)=>{
+        const msg = response.body.msg
+        expect(msg).toBe("bad request")
+      })
+ })
```
Comment (robert112233) [inline __tests__/app.test.js:141]: You are missing 2 sad path tests. 

400 - What if I send the body without the correct keys? Eg. { notusername: "philippaclaire9", information: "some text"} 

404 - What if i send a post request to /api/reviews/3/comments but the username doesn't exist in the database? Eg. { username: "Rob", body: "Here is my comment"}

File: controller/controllers.erros.js [hunk 4]
```diff
@@ -5,7 +5,12 @@ const handel404PathError =(req, res, next)=>{
 const handelCustomErrors =(err, req, res, next)=>{
     if(err.status && err.msg){
         res.status(err.status).send({msg:err.msg})
-    }else{
+    }else if (err.code === "22P02"){
+       res.status(400).send({mgs:'bad request'})
+    }else if (err.code === '23502'){
+        res.status(404).send({msg:'not found'})
+    }
+    else {
```
Comment (robert112233) [inline controller/controllers.erros.js:13]: Good error handling middleware 👍

**Review by robert112233 — CHANGES_REQUESTED** [review]
Good work, see my comments then submit another PR for review 👍

**Pull request merged**